specs/image-update-monitoring.md

```diff
@@ -0,0 +1,32 @@
+# Docker Image Update Monitoring
+
+## Summary
+
+Many GitOps users would like to automate Kubernetes manifest changes in the deployment repository
+(see [Deployment Repo Update Automation](./deployment-repo-update.md)). The changes might be triggered by
+the CI pipeline run or a new image in the Docker registry. The [Flux] provides the docker registry monitoring as part of
+[Automated Image Update](https://docs.fluxcd.io/en/latest/references/automated-image-update.html) feature but that functionality
+is not available for non-Flux users and cannot be leveraged in manifests are generated by a script (e.g. jsonnet).
+
+This document is meant to collect requirements for a component that provides docker registry monitoring functionality and 
+can be used by Argo CD and potentially Flux users.
+
+## Requirements
+
+### Configurable Event Handle
+
+When a new docker image is discovered the component should execute a pre-configured event handler and pass the docker image name/version as a parameter. The handler should be
+configured in a form of a shell script.
+
+### Docker Registry WebHooks
+
+Some Docker Registries send a webhook when a new image gets pushed. The component should provide a webhook handler which when invokes an event handler.
+
+### Image Pulling
+
+In addition to the webhook, the component should support image pulling. The pulling should detect the new images and invoke an event handler for each new image.
+
+### Image Credentials Auto-Discovering
+
+If a component is running inside of a Kubernetes cluster together with the deployments then it already has access to the Docker registry credentials. Auto-Discovering functionality
```

Comment:
This sounds a bit like the discovery of credentials providers in Kubernetes itself. :)